src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -0,0 +1,413 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Formats.Tar
+{
+    /// <summary>
+    /// Abstract class that represents a tar entry from an archive.
+    /// </summary>
+    /// <remarks>All the properties exposed by this class are supported by the <see cref="TarFormat.V7"/>, <see cref="TarFormat.Ustar"/>, <see cref="TarFormat.Pax"/> and <see cref="TarFormat.Gnu"/> formats.</remarks>
+    public abstract partial class TarEntry
+    {
+        internal TarHeader _header;
+        // Used to access the data section of this entry in an unseekable file
```

Comment:
Nit: For such comments, my preference is to use XML comments.  They don't add a lot of text, and they make IntelliSense more helpful when working in the project.

Reply:
Sure, I can change them.